pkg/controller/sparkapplication/controller.go

```diff
@@ -462,6 +458,9 @@ func shouldRetry(app *v1beta2.SparkApplication) bool {
 		if app.Spec.RestartPolicy.Type == v1beta2.Always {
 			return true
 		}
+		if app.Spec.Mode == v1beta2.ClientMode && strings.Contains(app.Status.AppState.ErrorMessage, "exceeded quota") && app.Status.SubmissionAttempts <= 14 {
```

Comment:
you should not hard code 14 here ... the max submission attempts should come from the CRD

Reply:
was going to do submissions from crd but if there is restart policy is set to NEVER then onSubmissionFailureRetries& 
onSubmissionFailureRetryInterval will not be present and cause nll pointer and operator to crash

Reply:
but do need a var for it

Reply:
If the restartPolicy is set to never, its fine to not retry on submission failures and go to Failing/Failed state

Reply:
okay sounds good will make that change then